derp/derphttp/derphttp_test.go

```diff
@@ -323,6 +323,12 @@ func TestBreakWatcherConnRecv(t *testing.T) {
 
 		timer.Reset(5 * time.Second)
 	}
+	// Closing the DERP client and waiting for the watch connection loop goroutine to finish
+	// is primarily here to avoid data races between the watch goroutine and test teardown
+	// (restoring retryInterval).
+	watcher1.Close()
+	cancel()
+	wg.Wait()
```

Comment:
can't you instead just move lines 285-288 up earlier, to the very of the Test func before the "var wg"? You already defer clean up all this stuff.